Grid/qcd/utils/GaugeGroupTwoIndex.h

```diff
@@ -31,25 +31,93 @@ enum TwoIndexSymmetry { Symmetric = 1, AntiSymmetric = -1 };
 
 inline Real delta(int a, int b) { return (a == b) ? 1.0 : 0.0; }
 
+template <int nc, TwoIndexSymmetry S, class group_name>
+struct DimensionHelper;
```

Comment:
main motivation would be the future implementation of SO(2N) but I don't have strong opinions on this

Reply:
Are you aware of concrete plans to implement this in the near future?

Reply:
no!

Reply:
Then, keep it simple, I'd say. Otherwise for the next ten years, people looking at this code will have to read understand and potentially change 20 lines of code instead of one.